### PR TITLE
move devfile build commands into standalone script

### LINKED ISSUE #90: Build command is too verbose
It's be easier to visually scan the build command in the form of:

`build-cmd.sh 20.0.0.10`

With output whether it runs fully or sees the marker file and stops.

### PR DESCRIPTION
**Pull request by ajm01**
Signed-off-by: Andrew J. Mauer <ajmauer@gmail.com>

Fixes #90, #37

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -41,7 +41,7 @@ generate() {
     mkdir -p generated
     sed -e "s!{{.OL_RUNTIME_VERSION}}!$OL_RUNTIME_VERSION!; s!{{.STACK_IMAGE}}!$STACK_IMAGE!; s!{{.DEVFILE_DOCKERFILE_LOC}}!$DEVFILE_DOCKERFILE_LOC!; s!{{.DEVFILE_DEPLOY_YAML_LOC}}!$DEVFILE_DEPLOY_YAML_LOC!" templates/devfile.yaml > generated/devfile.yaml
     sed -e "s!{{.BASE_OS_IMAGE}}!$BASE_OS_IMAGE!; s!{{.OL_RUNTIME_VERSION}}!$OL_RUNTIME_VERSION!" templates/stackimage/Dockerfile > generated/stackimage-Dockerfile
```
Comment (scottkurz) [inline build.sh:43]: Don't we need to bump up the stack image tag that the devfile uses if we're rebuilding the image?

**Review by scottkurz — CHANGES_REQUESTED** [review]
See my comment.. I could be wrong.. just trying to remember.

Also.. can we add procps via yum package to our stack image ..can be handy.

File: build.sh (hunk 1 above)
Comment (ajm01) [reply build.sh:43]: yes, it should be bumped, will do - and will add procps

File: stackimage/scripts/devbuild-cmd.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+if [ -e /projects/.disable-bld-cmd ];
+then
+    echo "found the disable file" && echo "devBuild command will not run" && exit 0;
+else
+    if [ ! -e /projects/pom.xml ];
+	then 
+             echo
+             echo "  pom.xml not found, non-viable or empty maven project. Please add src code and re-push";
```
Comment (scottkurz) [inline stackimage/scripts/devbuild-cmd.sh:10]: How about a bit more of an eye-catcher:
>  "ERROR:  pom.xml not ...."

File: templates/stackimage/Dockerfile [hunk 3]
```diff
@@ -8,7 +8,7 @@ RUN  useradd --uid 1001 --gid 0 --shell /bin/bash --create-home java_user
 
 # Dependency install
 RUN yum install --disableplugin=subscription-manager -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm \
-   && yum install --disableplugin=subscription-manager -y unzip curl ca-certificates wget xmlstarlet
+   && yum install --disableplugin=subscription-manager -y unzip curl ca-certificates wget xmlstarlet procps
```
Comment (scottkurz) [inline templates/stackimage/Dockerfile:11]: thanks

File: build.sh [hunk 4]
```diff
@@ -41,7 +41,7 @@ generate() {
...
-
+
```
Comment (scottkurz) [inline build.sh:44]: You didn't update the stack image yet.

**Review by scottkurz — CHANGES_REQUESTED** [review]

File: templates/devfile.yaml [hunk 5]
```diff
@@ -1,7 +1,7 @@
 schemaVersion: 2.0.0
 metadata:
   name: java-openliberty
-  version: 0.4.1
+  version: 0.4.3
```
Comment (scottkurz) [inline templates/devfile.yaml:4]: Did we skip 0.4.2?

File: templates/devfile.yaml [hunk 6]
```diff
@@ -76,7 +64,7 @@ commands:
     # The 'test' command requires an active container, so we don't need to specify the liberty runtime version
     exec:
       component: devruntime 
-      commandLine: mvn -Dmicroshed_hostname=localhost -Dmicroshed_http_port=9080 -Dmicroshed_manual_env=true -Dmicroshed_app_context_root=/ failsafe:integration-test failsafe:verify
+      commandLine: mvn -DforkCount=0 -Dmicroshed_hostname=localhost -Dmicroshed_http_port=9080 -Dmicroshed_manual_env=true -Dmicroshed_app_context_root=/ failsafe:integration-test failsafe:verify
```
Comment (scottkurz) [inline templates/devfile.yaml:67]: Ah you're not just working around in the regression test.  Hmm...let me just look into that a bit.

**Review by scottkurz — APPROVED** [review]

**Pull request merged**